src/new/wallet.js

```diff
@@ -2355,34 +2355,12 @@ class HathorWallet extends EventEmitter {
    * }>} Input and signature information
    */
   async getSignatures(tx, { pinCode = null } = {}) {
-    if (await this.storage.isReadonly()) {
-      throw new WalletFromXPubGuard('getSignatures');
-    }
     const pin = pinCode || this.pinCode;
-    if (!pin) {
-      throw new Error(ERROR_MESSAGE_PIN_REQUIRED);
-    }
-
-    const xprivkey = await this.storage.getMainXPrivKey(pin);
-    const key = HDPrivateKey(xprivkey);
-
-    const signatures = [];
-    const dataToSignHash = tx.getDataToSignHash();
-    for (const indexes of await this.getWalletInputInfo(tx)) {
-      const { addressIndex } = indexes;
-      // Derive key to addressIndex
-      const derivedKey = key.deriveNonCompliantChild(addressIndex);
-      const privateKey = derivedKey.privateKey;
-      // Get tx signature and populate transaction
-      const sigDER = transactionUtils.getSignature(dataToSignHash, privateKey);
-      signatures.push({
-        signature: sigDER.toString('hex'),
-        pubkey: privateKey.publicKey.toString(),
-        ...indexes,
-      });
-    }
-
-    return signatures;
+    const signatures = await this.storage.signTxP2PKH(tx, pin);
+    return signatures.map(sigData => ({
+      ...sigData,
+      addressPath: this.getAddressPathForIndex(sigData.addressIndex),
+    }));
```

Comment:
note: This will allow us to sign transactions for P2SH wallets using either the bitcore signing from the storage mainKey or the custom signing method.